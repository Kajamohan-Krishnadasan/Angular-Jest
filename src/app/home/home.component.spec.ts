import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { CapitalizePipe } from '../shared/capitalize.pipe';
import { HomeComponent } from './home.component';
import { HomeService } from './home.service';
import { HighlightDirective } from '../shared/highlight.directive';
import { capitalize } from 'lodash';

const testData = [
  {
    name: 'italian',
    image: 'aaron-burden-3UoB1ftLJjk-unsplash.jpg',
    alt: 'Italian Trulli',
  },
  {
    name: 'sri lanka',
    image: '4k-ultra-hd-motivational-focus-bs4nnk5bt18sgwi4.jpg',
    alt: 'sri lanka',
  },
];

class HomeServiceStub {
  getCities() {
    return Promise.resolve(testData);
  }
}

describe('HomeComponent', () => {
  let fixture: ComponentFixture<HomeComponent>;
  let homeComponent: HomeComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [HomeComponent, CapitalizePipe, HighlightDirective],
    }).overrideComponent(HomeComponent, {
      set: {
        providers: [
          {
            provide: HomeService,
            useClass: HomeServiceStub,
          },
        ],
      },
    });

    fixture = TestBed.createComponent(HomeComponent);
    homeComponent = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('is created and data received', waitForAsync(() => {
    fixture.whenStable().then(() => {
      expect(homeComponent).toBeDefined();
      expect(homeComponent.cities).toBe(testData);
    });
  }));

  it('verify html', waitForAsync(() => {
    fixture.whenStable().then(() => {
      fixture.detectChanges();

      // snapshot
      expect(fixture).toMatchSnapshot();

      const displayElements = fixture.debugElement.queryAll(By.css('p'));
      displayElements.forEach((element, index) => {
        expect(element.nativeElement.textContent).toContain(
          capitalize(testData[index].name)
        );
      });

      const imageElements = fixture.debugElement.queryAll(By.css('img'));

      imageElements.forEach((element, index) => {
        expect(element.nativeElement.src).toContain(testData[index].image);
        expect(element.nativeElement.alt).toContain(testData[index].alt);
      });
    });
  }));
});
