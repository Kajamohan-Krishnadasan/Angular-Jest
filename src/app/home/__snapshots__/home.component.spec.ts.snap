// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent verify html 1`] = `
<app-home
  cities={[Function Array]}
  homeService={[Function HomeServiceStub]}
>
  <ul>
    <li>
      <p
        apphighlight="yellow"
        style="background-color: yellow;"
      >
         Italian 
      </p>
      <img
        alt="Italian Trulli"
        src="assets/aaron-burden-3UoB1ftLJjk-unsplash.jpg"
      />
    </li>
    <li>
      <p
        apphighlight="yellow"
        style="background-color: yellow;"
      >
         Sri lanka 
      </p>
      <img
        alt="sri lanka"
        src="assets/4k-ultra-hd-motivational-focus-bs4nnk5bt18sgwi4.jpg"
      />
    </li>
  </ul>
</app-home>
`;
